core-bundle/src/Doctrine/Backup/Backup.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Doctrine\Backup;
+
+use Symfony\Component\Filesystem\Filesystem;
+use Webmozart\PathUtil\Path;
+
+class Backup
+{
+    public const DATETIME_FORMAT = 'YmdHis';
+    public const VALID_BACKUP_NAME_REGEX = '@^.*__(\d{4}\d{2}\d{2}\d{2}\d{2}\d{2})\.sql(\.gz)?$@';
+
+    private string $filepath;
+    private \DateTimeInterface $createdAt;
+
+    public function __construct(string $filepath)
+    {
+        $this->filepath = self::validateFilePath($filepath);
+        $this->createdAt = self::extractDatetime($filepath);
+    }
+
+    /**
+     * Returns the size in bytes.
+     */
+    public function getSize(): int
+    {
+        if (!(new Filesystem())->exists($this->getFilepath())) {
+            return 0;
+        }
+
+        return (int) filesize($this->getFilepath());
+    }
+
+    public function getFilepath(): string
+    {
+        return $this->filepath;
+    }
+
+    public function getCreatedAt(): \DateTimeInterface
+    {
+        return $this->createdAt;
+    }
+
+    public static function createNewAtPath(string $targetPath, \DateTime $dateTime = null): self
+    {
+        $targetPath = rtrim($targetPath, '/');
+        $now = $dateTime ?? new \DateTime('now');
+        $now->setTimezone(new \DateTimeZone('UTC'));
+        $filepath = sprintf('%s/backup__%s.sql.gz', $targetPath, $now->format(self::DATETIME_FORMAT));
+
+        (new Filesystem())->dumpFile($filepath, '');
+
+        return new self($filepath);
+    }
+
+    public function toArray(): array
+    {
+        return [
+            'createdAt' => $this->getCreatedAt()->format(\DateTimeInterface::ISO8601),
```

Comment:
<img width="786" alt="" src="https://user-images.githubusercontent.com/1192057/143050780-278e39e6-eb0f-48ae-856a-8b4563d5a354.png">

Can this be ignored or should we change it?

Reply:
We should probably use ATOM, yeah. You can change that, no problem :)

Reply:
I have added the changes to [Toflar#11](https://github.com/Toflar/contao/pull/11).